Libraries/src/Amazon.Lambda.AspNetCoreServer/APIGatewayProxyFunction.cs

```diff
@@ -154,12 +154,21 @@ protected override void MarshallRequest(InvokeFeatures features, APIGatewayProxy
                 requestFeatures.Method = apiGatewayRequest.HttpMethod;
 
                 string path = null;
+
+                // Replaces {proxy+} in path, if exists
                 if (apiGatewayRequest.PathParameters != null && apiGatewayRequest.PathParameters.TryGetValue("proxy", out var proxy) &&
                     !string.IsNullOrEmpty(apiGatewayRequest.Resource))
                 {
                     var proxyPath = proxy;
                     path = apiGatewayRequest.Resource.Replace("{proxy+}", proxyPath);
+
+                    // Adds all the rest of non greedy parameters in apiGateway.Resource to the path
+                    foreach (var pathParameter in apiGatewayRequest.PathParameters.Where(pp => pp.Key != "proxy"))
+                    {
+                        path = path.Replace($"{{{pathParameter.Key}}}", pathParameter.Value);
+                    }
                 }
+                // As a matter of fact, is there a reason for the if clause above? A simpler/faster solution would be to straight on replace the path as below
 
```

Comment:
I think all the logic, between lines 159 and 169 is not needed at all. 
If that logic is removed, both {proxy+} and non proxy paths will be replaced just fine on line 175.

In fact this whole block of code:
`string path = null;
if (apiGatewayRequest.PathParameters != null && apiGatewayRequest.PathParameters.TryGetValue("proxy", out var proxy) &&
    !string.IsNullOrEmpty(apiGatewayRequest.Resource))
{
    var proxyPath = proxy;
    path = apiGatewayRequest.Resource.Replace("{proxy+}", proxyPath);
}

if (string.IsNullOrEmpty(path))
{
    path = apiGatewayRequest.Path;
}`

should just become:
`string path = apiGatewayRequest.Path;`

This would fix the issue, and be a lot cleaner